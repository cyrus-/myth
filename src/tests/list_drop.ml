let s = "
type nat =
  | O
  | S of nat

type list =
  | Nil
  | Cons of nat * list

let list_drop : list -> nat -> list |>
  { [] => ( 0 => []
          | 1 => []
          | 2 => [] )
  | [0] => ( 0 => [0]
           | 1 => [] )
  | [1] => ( 0 => [1]
           | 1 => [] )
  | [1; 0] => ( 0 => [1; 0]
              | 1 => [0]
              | 2 => [] )
  | [0; 1] => ( 0 => [0; 1]
              | 1 => [1]
              | 2 => [] )
  } = ?"
