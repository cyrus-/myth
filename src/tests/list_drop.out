(* Generated Thu Nov 13 00:35:23 2014 *)
let list_drop : list -> nat -> list =
  let rec f1 (l1:list) : nat -> list =
    fun (n1:nat) ->
      match n1 with
        | O -> l1
        | S (n2) -> (match l1 with
                       | Nil -> Nil
                       | Cons (n3, l2) -> f1 l2 n2)
  in
    f1
;;
